src/compiler/parser.ts

```diff
@@ -5861,6 +5879,13 @@ namespace ts {
                 return parseAccessorDeclaration(<AccessorDeclaration>node, SyntaxKind.SetAccessor);
             }
 
+            if (token() === SyntaxKind.StringLiteral) {

```

Comment:
Why not just check the token text here and call parseConstructorDeclaration, and have parseConstructorDeclaration handle whether its a string or keyword? The public fields proposal disallows public fields with the name `constructor` or `"constructor"`.

Reply:
Just because we already give a better error at the grammar checking stage for fields named `constructor`—it says `Classes may not have a field named 'constructor'.` instead of `'(' expected`.